sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py

```diff
@@ -233,12 +250,62 @@ def _transmit(self, envelopes: List[TelemetryItem]) -> ExportResult:
                                     error.message,
                                     envelopes[error.index] if error.index is not None else "",
                                 )
+                            # Track dropped items in customer statsbeat
+                            if self._customer_statsbeat_metrics and error.index is not None:
+                                try:
+                                    telemetry_type = self._get_telemetry_type(envelopes[error.index])
+                                    self._customer_statsbeat_metrics.count_dropped_items(
+                                        1,
+                                        DropCode.NON_RETRYABLE_STATUS_CODE,
+                                        telemetry_type
+                                    )
+                                except Exception as track_ex:
+                                    logger.warning("Failed to track dropped items in customer statsbeat: %s", track_ex)
+                        
                     if self.storage and resend_envelopes:
+                        # Track retried items in customer statsbeat
+                        if self._customer_statsbeat_metrics and resend_envelopes:
+                            try:
+                                for envelope in resend_envelopes:
+                                    telemetry_type = self._get_telemetry_type(envelope)
+                                    self._customer_statsbeat_metrics.count_retry_items(
+                                        1,
+                                        RetryCode.RETRYABLE_STATUS_CODE,
```

Comment:
Similar to the comment about, the retry code is not the string but the actual status code.